lightning/src/ln/onchaintx.rs

```diff
@@ -141,6 +152,12 @@ macro_rules! subtract_high_prio_fee {
 /// do RBF bumping if possible.
 pub struct OnchainTxHandler<ChanSigner: ChannelKeys> {
 	destination_script: Script,
+	funding_redeemscript: Script,
+	local_commitment: Option<Transaction>,
```

Comment:
I'm admittedly pretty dissapointed in this setup - part of the goal here is to make it easy to ensure that we only ever create signatures for the latest local commitment transaction/one set of HTLC transactions. The way to do this would be to lean on the type system and encapsulation - make some struct that contains the latest and prev LocalCommitmentTransactions and their HTLCs, then have some method which gets a signed copy of at most one of them, and then refuses to ever be updated. Moving to `Transaction`s directly seems like a step backwards, not forwards.